tests/ha/sleep.pm

```diff
@@ -0,0 +1,22 @@
+# SUSE's openQA tests
+#
+# Copyright (c) 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Sleep after reboot
+# Maintainer: Ricardo Branco <rbranco@suse.com>
+
+use base "basetest";
+use strict;
+use warnings;
+
+sub run {
+    # Sleep 5 minutes to let the slave become the master
+    sleep 300;
```

Comment:
Is it very useful to create a new test file only for this? Couldn't be added in the needed test (and conditioned to HANA test of course)?

Reply:
Agree with @ldevulder. I think this code can be added at the end of `ha/fencing` by checking `if is_sles4sap and check_var('AUTOMATED_REGISTER', 'false');`

Reply:
> Is it very useful to create a new test file only for this? Couldn't be added in the needed test (and conditioned to HANA test of course)?

No because I have to sleep here after reboot and before `boot_to_desktop` in the `AUTOMATED_REGISTER=true` scenario.  It's a way of controlled shutdown.

Reply:
> No because I have to sleep here after reboot and before `boot_to_desktop` in the `AUTOMATED_REGISTER=true` scenario. It's a way of controlled shutdown.

I really don't know why this:
`loadtest 'ha/sleep' if (check_var('AUTOMATED_REGISTER', 'false') && get_var('HA_CLUSTER_INIT'));`
couldn't be replaced by this:
`sleep 300 if (check_var('AUTOMATED_REGISTER', 'false') && get_var('HA_CLUSTER_INIT'));`
at the end of `ha/fencing.pm` (maybe with the use of `bmwqemu::scale_timeout` also).

But maybe I missed something

Reply:
I'm really, really against having a module within the scope of `ha/` that only does a `sleep()` as that's not something HA-specific.

If this `sleep` code cannot be added to `ha/fencing` as described by both me and @ldevulder and the only way to have a controlled shutdown is with an extra module, this should go elsewhere IMO.

So once more:

1. This should go into the code of `ha/fencing`, at the very end, after all `barrier_wait()` calls, and as `sleep bmwqemu::scale_timeout(300) if (check_var('AUTOMATED_REGISTER', 'false') && get_var('HA_CLUSTER_INIT') && is_sles4sap);`
2. If this doesn't work as expected, and it is required to have a "sleep module" to have a controlled shutdown, move this to an appropriate directory within tests, as this sleep module is not HA specific. Hints: sles4sap or boot.